R/analyse_functional_network.R

```diff
@@ -147,36 +165,40 @@ does not match the number of rows in your data.", prefix = "\n", initial = ""))
   payload_id <- NULL
 
   if (!missing(binds_treatment)) {
-    if (missing(halo_color)) {
-      coloring <- input %>%
-        dplyr::filter({{ binds_treatment }}) %>%
-        dplyr::mutate(color = "#5680C1")
-    } else {
-      coloring <- input %>%
-        dplyr::filter({{ binds_treatment }}) %>%
-        dplyr::mutate(color = halo_color)
+    if (is.null(halo_color)) {
+      halo_color <- "#5680C1"
     }
+
+    coloring <- input %>%
+      dplyr::filter({{ binds_treatment }}) %>%
+      dplyr::mutate(color = halo_color)
+
     payload_id <- string_db$post_payload(dplyr::pull(coloring, {{ string_id }}),
       colors = coloring$color
     )
   }
-  if (plot == TRUE) {
-    if (length(unique(dplyr::pull(data, !!ensym(protein_id)))) > 400) {
-      stop(strwrap("Please only provide the top 400 significant proteins for plots! String
-cannot plot more at once.", prefix = "\n", initial = ""))
+
+
+  tryCatch(
```

Comment:
I just noticed that the `return()` at the end doesn't return from the overall function if an error occurs but just from the `try_catch()`. So you cannot actually use it like this within the function. Only works because it is at the end. This is important for the comment above. 
My solution would the following. Maybe you have another idea.
```
interactions <- tryCatch(
    {
      if (plot) {
        string_db$plot_network(string_ids, payload_id = payload_id)
      } else {
        mapping <- input %>%
          dplyr::distinct({{ protein_id }}, {{ string_id }})

        string_db$get_interactions(string_ids) %>%
          dplyr::left_join(mapping, by = c("from" = rlang::as_name(rlang::enquo(string_id)))) %>%
          dplyr::rename(from_protein = {{ protein_id }}) %>%
          dplyr::left_join(mapping, by = c("to" = rlang::as_name(rlang::enquo(string_id)))) %>%
          dplyr::rename(to_protein = {{ protein_id }}) %>%
          dplyr::distinct()
      }
    },
    error = function(e) {
      e$message
    }
  )

  if (is.character(interactions)) {
    message("An error occurred during the interaction network analysis: ", interactions)
    return(invisible(NULL))
  } else {
    return(interactions)
  }
```